trussworks/react-uswds
# feat: Icon component using SVGR

# Summary

## Related Issues or PRs

This PR implements the USWDS Icon component using an alternative approach, converting each icon to inline SVGs that can be rendered as React components instead of a single `sprite.svg` file with the `<use>` element. I worked off the existing PR (https://github.com/trussworks/react-uswds/pull/1241/files) and used most of the existing `Icon` component code in an HOC that wraps each React SVG component with the supported props (ie, `size`) and adds the USWDS classes.

Closes #992 

## How To Test

* View the Icons in the deployed Storybook for this branch and verify they match the icons listed at [USWDS](https://designsystem.digital.gov/components/icon/). Note that the thumbs down icon changed in the [2.11.2 release](https://github.com/uswds/uswds/releases/tag/v2.11.2), so that asset will appear differently until we've updated to that version of USWDS.
* Check out the branch locally and start the Example app, and view the Icons page. It should display all of the new icons.
* If you have another app that uses this library, try installing this branch and verify you can import and render icon components in your app.

### Screenshots (optional)

![image](https://user-images.githubusercontent.com/2723066/135172715-579aa270-75e8-440d-9a03-bbf6735106b3.png)
